subprojects/core/src/main/java/org/gradle/caching/local/internal/LocalBuildCacheService.java

```diff
@@ -0,0 +1,33 @@
+/*
+ * Copyright 2017 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.caching.local.internal;
+
+import org.gradle.api.Action;
+import org.gradle.caching.BuildCacheKey;
+
+import java.io.Closeable;
+import java.io.File;
+
+public interface LocalBuildCacheService extends Closeable {
+
+    void store(BuildCacheKey key, File file);
+
+    void load(BuildCacheKey key, Action<? super File> reader);
```

Comment:
I've done this. I originally didn't because we don't always have a local cache, but I worked around that with a fallback.